### ISSUE TITLE
[Tutorial] Potentially confusing explanation of how a component rendering is triggered

### ISSUE THREAD

**Issue by pubudu91** [reactjs/reactjs.org#3894]
I'm new to React and when learning, I first went through the tutorial and then went through the docs. In the tutorial I didn't quite understand why we absolutely have to use the `setState()` method. But when going through the [Using State Correctly](https://reactjs.org/docs/state-and-lifecycle.html#using-state-correctly) and then the [API docs for `setState()`](https://reactjs.org/docs/react-component.html#setstate) I understood why. Then I went back to the tutorial to see why I couldn't get this from the tutorial. State is introduced in the [Making an Interactive Component](https://reactjs.org/tutorial/tutorial.html#making-an-interactive-component) section. 

I think it'd be better if it's explicitly mentioned here that state changes are what causes components to be re-rendered and hence, one shouldn't attempt to modify `state` directly. The following part particularly is what gave the wrong impression I think. I took this to mean that it's events such as `onClick` that triggers a rendering. 

> By calling `this.setState` from an `onClick` handler in the Square’s `render` method, we tell React to re-render that Square whenever its `<button>` is clicked.

**Comment by premsai2030** [comment]
Yes you were absolutely correct, I think you have done with class components till now. Try with functional components as well. Really the hooks concept was so good....
